plugins/woocommerce/src/Blocks/Templates/AbstractTemplate.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * AbstractTemplate class.
+ *
+ * Shared logic for templates.
+ *
+ * @internal
+ */
+abstract class AbstractTemplate {
+
+	/**
+	 * The slug of the template.
+	 *
+	 * @var string
+	 */
+	const SLUG = '';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
```

Comment:
Same for templates, I think we should invoke the init method in the block registry instead.

Reply:
Done in 0b5e0d6b163bea818d8547cfba9e497bc27803b0.